### PR TITLE
Apb 5023 updates des stub

### PR DESCRIPTION
**Pull request by Leeroyc720**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsexternalstubs/models/RegexPatterns.scala [hunk 1]
```diff
@@ -31,6 +32,8 @@ object RegexPatterns {
     "^((?!(BG|GB|KN|NK|NT|TN|ZZ)|(D|F|I|Q|U|V)[A-Z]|[A-Z](D|F|I|O|Q|U|V))[A-Z]{2})\\s?\\d{2}\\s?\\d{2}\\s?\\d{2}\\s?[A-D]?$".r)
   val validArn: Matcher = validate("^[A-Z]ARN[0-9]{7}$".r)
   val validUtr: Matcher = validate("^[0-9]{10}$".r)
+  val validUrn: Matcher = validate("^([A-Z0-9]{1,15})$".r)
+  val validUtrOrUrn = validateTrustTaxIdentifier("^[0-9]{10}$".r, "^([A-Z0-9]{1,15})$".r)
```
Comment (geoffreywatson) [inline app/uk/gov/hmrc/agentsexternalstubs/models/RegexPatterns.scala:36]: where do we use it? not sure we need this

File: it/uk/gov/hmrc/agentsexternalstubs/controllers/DesStubControllerISpec.scala [hunk 2]
```diff
@@ -227,6 +229,171 @@ class DesStubControllerISpec
...
+                haveProperty[String]("trustName"),
+                haveProperty[String]("serviceName")
+          )
+        }
+      }
+
+    "GET /trusts/agent-known-fact-check/:urn"   should {
+      "respond 200 with trust details" in {
+        implicit val session: AuthenticatedSession = SignIn.signInAndGetSession()
+        val enrolmentKey = "HMRC-TERSNT-ORG~URN~0123456789"
```
Comment (geoffreywatson) [inline it/uk/gov/hmrc/agentsexternalstubs/controllers/DesStubControllerISpec.scala:383]: 0123456789 doesn't look like a URN value??

File: it/uk/gov/hmrc/agentsexternalstubs/controllers/ExampleDesPayloads.scala [hunk 3]
```diff
@@ -2,6 +2,21 @@ package uk.gov.hmrc.agentsexternalstubs.controllers
 
 trait ExampleDesPayloads {
 
+  val trustDetailsPayload =
+    """
+      |{
+      |  "acknowledgmentReference": "A1BCDEFG1HIJKLNOPQRSTUVWXYZ12346",
+      |   "refNumber": "XXTRUST80000001",
+      |   "agentReferenceNumber": "PARN0876123",
+      |   "idType": "UTR"
```
Comment (geoffreywatson) [inline it/uk/gov/hmrc/agentsexternalstubs/controllers/ExampleDesPayloads.scala:11]: URN?

File: app/uk/gov/hmrc/agentsexternalstubs/models/RegexPatterns.scala (hunk 1 above)
Comment (Leeroyc720) [reply app/uk/gov/hmrc/agentsexternalstubs/models/RegexPatterns.scala:36]: we use this in DESstubcontroler line 454

**Pull request merged**